### PR TITLE
Migrate Graql Delete ITs to BDD

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

Where possible, we would like our Graql `delete` tests to be migrated to BDD.

This PR forms part of the work towards #5719 .

## What are the changes implemented in this PR?

This PR will gradually delete test cases from GraqlDeleteIT as they are tested in BDD.

This PR should NOT be merged until verification is updated with the new BDD scenarios, which is the desired outcome of https://github.com/graknlabs/verification/pull/54.

### CODE REVIEW

File: test/integration/graql/query/GraqlDeleteIT.java [hunk 1]
```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
...
-        aRelation = tx.getConcept(relationId);
-        tx.getConcept(personId).delete();
-        assertTrue(aRelation.isDeleted());
-
-        assertEquals(aProvenance.owners().collect(Collectors.toList()).size(), 0);
-    }
-
-
-    @Test
-    public void afterDeletingAllInstances_TheTypeCanBeUndefined() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDeleteIT.java:295]: have we got this in `Undefine` already? Otherwise, we shouldn't delete it yet

File: test/integration/graql/query/GraqlDeleteIT.java [hunk 2]
```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
...
-        tx.execute(Graql.match(x.isa("fake-type"), y.isa("fake-type"), x.not(y.var()))
-                .delete(var("x").isa("fake-type"), var("y").isa("fake-type")));
-
-        assertNotExists(tx, var().isa("fake-type"));
-    }
-
-    @Test
-    public void whenDeletingAVariableNotInTheQuery_Throw() {
-        exception.expect(GraqlException.class);
-        exception.expectMessage(UNBOUND_DELETE_VARIABLE.getMessage(y.var()));
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDeleteIT.java:339]: is this documented as an error message to test in the future? Would be good to write down:

"* Test Unbound Delete Variable when variable in delete clause is not in match"

or something like this in your collection of messages to bring back tests for

File: test/integration/graql/query/GraqlDeleteIT.java [hunk 3]
```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
...
-        attrId = answers.get(0).get("a").id();
-
-        exception.expect(GraqlSemanticException.class);
-        exception.expectMessage("Cannot delete attribute ownership, concept [$a]");
-        exception.expectMessage("is not of required attribute type [thing]");
-        tx.execute(Graql.parse("match $x id " + ownerId + "; $x has title $a; $a id " + attrId + "; delete $x has thing $a;").asDelete());
-    }
-
-    @Test
-    public void whenDeletingAttributeOwnershipWithSubtype_Throw() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDeleteIT.java:405]: is this migrated? Just double check

File: test/integration/graql/query/GraqlDeleteIT.java [hunk 4]
```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
...
-        answers = tx.execute(Graql.parse("match $m isa movie, has title \"The Muppets\"; get;").asGet());
-        assertEquals(1, answers.size());
-        tx.execute(Graql.parse("match $m isa movie, has title \"The Muppets\"; delete $m isa thing;").asDelete());
-        answers = tx.execute(Graql.parse("match $m isa movie, has title \"The Muppets\"; get;").asGet());
-        assertEquals(0, answers.size());
-    }
-
-
-    @Test
-    public void deleteWithDirectType_ThrowsWhenNotDirectType() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDeleteIT.java:442]: we have the BDD test for direct type succeeding, do we have this negative case as well ?

File: test/integration/graql/query/GraqlDeleteIT.java [hunk 5]
```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
...
-        assertEquals(1, answers.size());
-        assertEquals(1, answers.get(0).get("r").asRelation().rolePlayers().count());
-    }
-
-    /*
-    We don't allow the following downcasted role in the delete clause:
-    match $r (super-role: $x) isa relation; delete $r (sub-role: $x);
-     */
-    @Test
-    public void whenDeletingRolePlayerAsSubRole_Throw() {
```
Comment (flyingsilverfin) [inline test/integration/graql/query/GraqlDeleteIT.java:530]: double check this has been migrated, only recall seeing that we don't throw when using the super-role

**Comment by alexjpwalker** [comment]
> Can you also add link to the PR where you added this into BDD, @alexjpwalker ? I will also assign jmsfltchr to review this first. I'll approve last.

Added, will follow the same pattern for other grakn BDD migration PRs

File: test/integration/graql/query/GraqlDeleteIT.java (hunk 1 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDeleteIT.java:295]: Thought I added it but apparently not! Will merge https://github.com/graknlabs/verification/pull/52 and then add it to undefine.feature.

Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDeleteIT.java:295]: ... OK, never mind, it was in master but not in my local repo somehow.

`undefine.feature` has a draft scenario named:
`Scenario: a type can be undefined after deleting all of its instances`

File: test/integration/graql/query/GraqlDeleteIT.java (hunk 2 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDeleteIT.java:339]: Yes, in https://github.com/graknlabs/grakn/issues/5750:

2. `match $x isa movie; delete $y isa thing;` throws "the delete clause variable [%s] is not defined in the match clause"

File: test/integration/graql/query/GraqlDeleteIT.java (hunk 3 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDeleteIT.java:405]: Yes and no - at the time that you wrote this comment, the scenario was in draft; I have since implemented it.

Furthermore, it should have been quick to verify that and it wasn't.

I'm updating my verification PR to group the delete.feature into subsections, named:
- Things
- Roleplayers
- Attribute ownerships
- Edge cases

File: test/integration/graql/query/GraqlDeleteIT.java (hunk 4 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDeleteIT.java:442]: Yes

File: test/integration/graql/query/GraqlDeleteIT.java (hunk 5 above)
Comment (alexjpwalker) [reply test/integration/graql/query/GraqlDeleteIT.java:530]: Again this scenario was in draft and I forgot to implement it. It has now been implemented!! You have a keen eye.

**Comment by alexjpwalker** [comment]
Closing due to polluted diff
